src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -19,38 +20,74 @@ internal class ManagedIdentityClient
     {
         private const string WindowsHimdsFilePath = "%Programfiles%\\AzureConnectedMachineAgent\\himds.exe";
         private const string LinuxHimdsFilePath = "/opt/azcmagent/bin/himds";
+
+        // Cache for the managed identity source
+        private static ManagedIdentitySource? s_cachedManagedIdentitySource;
+        private static readonly SemaphoreSlim s_credentialSemaphore = new(1, 1);
         private readonly AbstractManagedIdentity _identitySource;
 
-        public ManagedIdentityClient(RequestContext requestContext)
+        internal static async Task<ManagedIdentityClient> CreateAsync(RequestContext requestContext, CancellationToken cancellationToken = default)
```

Comment:
I believe the requestContext has the cancellation token, but it's ok to move it as last param. But I recommend you do not use a default.